app/services/remove_status_service.rb

```diff
@@ -22,7 +23,12 @@ def call(status)
 
     @status.destroy!
 
-    return unless @account.local?
+    # There is no reason to send out Undo activities when the
+    # cause is that the original object has been removed, since
+    # original object being removed implicitly removes reblogs
+    # of it. The Delete activity of the original is forwarded
+    # separately.
+    return if !@account.local? || @options[:original_removed]
```

Comment:
still don't know if I agree with this—some implementations might treat the deletion of the original vs deletions of reblogs separately

(for example, this is how it works on tumblr so sites looking to implement tumblr-like behavior where reblogs can add addition comments might not like it)

Reply:
I mean if you think about it in terms of user actions, nobody actually chooses to un-reblog a post that's been deleted. What we've been doing is enforcing a certain implicit behaviour explicitly. If your software doesn't have referential integrity and you want to display reblogs-of-dead-objects, maybe that's fine?

Given that both delete-forwarding and undo-announces go out to the same recipients, removing this behaviour instantly halves the required traffic.